pipelines/base/template-build/template-build.yaml

```diff
@@ -75,13 +87,33 @@ spec:
           workspace: workspace
         - name: registry-auth
           workspace: registry-auth
+    - name: prefetch-dependencies
+      when:
+      - input: $(params.is-hermetic)
+        operator: in
+        values: ["true"]
+      params:
+        - name: package-type
+          value: "$(params.prefetch-package-type)"
+        - name: package-path
+          value: "$(params.path-context)"
+        - name: flags
+          value: "$(params.prefetch-flags)"
+      runAfter:
+        - appstudio-configure-build
+      taskRef:
+        name: prefetch-dependencies
+        version: "0.1"
+      workspaces:
+        - name: source
+          workspace: workspace
     - name: build-container
       when:
       - input: $(tasks.appstudio-init.results.build)
         operator: in
         values: ["true"]
       runAfter:
-        - appstudio-configure-build
+        - prefetch-dependencies
```

Comment:
Word of warning on `when` expressions + `runAfter`: depending on the Tekton Pipelines version and the configuration of the Pipelines Controller, this may behave rather unexpectedly. If `when` evaluates to false and prefetch-dependencies is skipped, all the dependent tasks (`runAfter` is a dependency) may be skipped as well!

As far as I understand, skipping everything is the default in [v0.32](https://tekton.dev/vault/pipelines-v0.32.x/pipelines/#guarding-a-task-and-its-dependent-tasks)
Skipping only the task itself is the default in [v0.33](https://tekton.dev/vault/pipelines-v0.33.x/pipelines/#guarding-a-task-and-its-dependent-tasks)
And the config option is not mentioned anymore (skips only the single task) in [v0.35](https://tekton.dev/vault/pipelines-v0.35.x/pipelines/#guarding-a-task-and-its-dependent-tasks)

Not sure what versions HACBS supports, so this may be a non-issue. It made us sad in OSBS2 though :disappointed: